src/seeds.proposals.cpp

```diff
@@ -271,101 +275,125 @@ void proposals::onperiod() {
     cycle_table c = cycle.get_or_create(get_self(), cycle_table());
     uint64_t current_cycle = c.propcycle;
 
-    while (pitr != props.end()) {
-      if (pitr->stage == stage_active) {
-        votes_tables votes(get_self(), pitr->id);
-        uint64_t voters_number = distance(votes.begin(), votes.end());
+    std::vector<uint64_t> active_props;
+    std::vector<uint64_t> eval_props;
 
-        double majority = double(prop_majority) / 100.0;
-        double fav = double(pitr->favour);
-        bool passed = pitr->favour > 0 && fav >= double(pitr->favour + pitr->against) * majority;
-        name prop_type = get_type(pitr->fund);
-        bool is_alliance_type = prop_type == alliance_type;
-        bool is_campaign_type = prop_type == campaign_type;
+    if (props_by_status.begin() != props_by_status.end()) {
+      auto p1 = props_by_status.begin();
+      auto p2 = p1;
```

Comment:
@IanMendozaJaimes Are you sure this is going to work - that p2 won't be changed when p1 changes due to changes to status in the props list?

How I would solve this is to iterate over status eval and capture the IDs, then iterate over status open and capture those IDs, then iterate over the list of Ids captured and find proposals by ID. 

Less efficient, admittedly, but no pointer magic. Anwyay we shall see I hope it works like this ;)